Public/Src/Cache/ContentStore/InterfacesTest/FileSystem/MemoryFileSystem.cs

```diff
@@ -162,7 +162,7 @@ protected virtual void Dispose(bool disposing)
         }
 
         /// <inheritdoc />
-        public IEnumerable<FileInfo> EnumerateFiles(AbsolutePath path, EnumerateOptions options)
+        public virtual IEnumerable<FileInfo> EnumerateFiles(AbsolutePath path, EnumerateOptions options)

```

Comment:
>public [](start = 8, length = 6)

What is reason for making virtual?

Reply:
The mock filesystem inherits from `MemoryFileSystem` in order to avoid having to implement everything it does. The function is required to be virtual to be able to override it.

Reply:
By the way, your answers are showing up as separate comments rather than as threaded answers to my comments. I assume you are using the github web interface. Is it possible to get threaded comments in that interface?

---
In reply to: [285271628](https://github.com/microsoft/BuildXL/pull/329#discussion_r285271628) [](ancestors = 285271628)